apisix/debug.lua

```diff
@@ -123,15 +124,35 @@ local function apple_new_fun(module, fun_name, file_path, hook_conf)
 
     function mt.__call(self, ...)
         local arg = {...}
+        local http_filter = debug_yaml.http_filter
+        local api_ctx = ngx.ctx.api_ctx
+        local enable_by_hook = not (http_filter and http_filter.enable)
+                or (api_ctx and api_ctx.enable_dynamic_debug)
```

Comment:
We don't need this part. When not http_filter, there won't be api_ctx.enable_dynamic_debug too